ocaml-lsp-server/src/code_actions/action_inline.ml

```diff
@@ -0,0 +1,119 @@
+open Import
+
+let action_title = "Inline"
+
+type inline_task =
+  { ident : Ident.t
+  ; body : Typedtree.expression  (** the expression to inline *)
+  ; context : Typedtree.expression  (** where to perform inlining *)
+  }
+
+let find_inline_task pipeline pos =
+  let contains loc pos =
+    match Position.compare_inclusion pos (Range.of_loc loc) with
+    | `Outside _ -> false
+    | `Inside -> true
+  in
+
+  (* Find most enclosing nonrecursive let binding *)
+  let browse =
+    Mpipeline.typer_result pipeline
+    |> Mtyper.get_typedtree |> Mbrowse.of_typedtree
+  in
+
+  Mbrowse.enclosing
+    (Mpipeline.get_lexing_pos pipeline (Position.logical pos))
+    [ browse ]
+  |> List.find_map ~f:(function
+       | ( _
+         , Browse_raw.Expression
+             { exp_desc =
+                 Texp_let
+                   ( Nonrecursive
+                   , [ { vb_pat = { pat_desc = Tpat_var (id, s); _ }
+                       ; vb_expr
+                       ; _
+                       }
+                     ]
+                   , rhs )
+             ; _
+             } )
+         when contains s.loc pos ->
+         Some { ident = id; body = vb_expr; context = rhs }
+       | _ -> None)
+
+let find_parsetree_loc pipeline loc k =
```

Comment:
I think that offering inlining for functions is an important feature. Functions are first class, after all. That said, unparsing has some downsides. It obviously destroys comments (which is probably fine) and it inserts `[@merlin_loc]` annotations. 

Do you know if the merlin parsetree is from before or after ppx expansion? If it's from before expansion, filtering out the merlin annotations is a decent solution.

Your proposed implementation looks equivalent. Iteration doesn't continue after the call to `k`, because there is no call back into the iterator in that branch.

Reply:
I've updated the implementation to remove `[@merlin_loc]` annotations. All other ppx extensions are preserved.